targets/lenovo-x1-carbon.nix

```diff
@@ -0,0 +1,181 @@
+            icon = "${pkgs.weston}/share/weston/icon_editor.png";
+          }
+
+          {
+            path = "${pkgs.waypipe}/bin/waypipe ssh -i ${pkgs.waypipe-ssh}/keys/waypipe-ssh -o StrictHostKeyChecking=no 192.168.101.6 gala --enable-features=UseOzonePlatform --ozone-platform=wayland";
+            icon = "${pkgs.weston}/share/weston/icon_editor.png";
+          }
+
+          {
+            path = "${pkgs.waypipe}/bin/waypipe ssh -i ${pkgs.waypipe-ssh}/keys/waypipe-ssh -o StrictHostKeyChecking=no 192.168.101.7 zathura";
+            icon = "${pkgs.weston}/share/weston/icon_editor.png";
+          }
+        ];
+      })
+    ];
+    hostConfiguration = lib.nixosSystem {
+      inherit system;
+      specialArgs = {inherit lib;};
+      modules =
+        [
+          microvm.nixosModules.host
+          ../modules/host
+          ../modules/virtualization/microvm/microvm-host.nix
+          ../modules/virtualization/microvm/netvm.nix
+          ../modules/virtualization/microvm/guivm.nix
+          ../modules/virtualization/microvm/appvm.nix
+          ({
+            pkgs,
+            lib,
+            ...
+          }: {
+            services.udev.extraRules = ''
+              # Add usb to kvm group
+              SUBSYSTEM=="usb", ATTR{idVendor}=="046d", ATTR{idProduct}=="c52b", GROUP+="kvm"
+            '';
+            ghaf = {
+              hardware.x86_64.common.enable = true;
+
+              virtualization.microvm-host.enable = true;
+              host.networking.enable = true;
+              virtualization.microvm.netvm = {
+                enable = true;
+                extraModules = netvmExtraModules;
+              };
+              virtualization.microvm.guivm = {
+                enable = true;
+                extraModules = guivmExtraModules;
+              };
+              virtualization.microvm.appvm = {
+                enable = true;
+                vms = [
+                  {
+                    name = "chromium";
+                    packages = [pkgs.chromium];
+                    ipAddress = "192.168.101.5/24";
+                    macAddress = "02:00:00:03:03:05";
+                    ramMb = 3072;
+                    cores = 4;
+                  }
+                  {
+                    name = "gala";
+                    packages = [pkgs.gala-app];
+                    ipAddress = "192.168.101.6/24";
+                    macAddress = "02:00:00:03:03:06";
+                    ramMb = 1536;
+                    cores = 2;
+                  }
+                  {
+                    name = "zathura";
+                    packages = [pkgs.zathura];
+                    ipAddress = "192.168.101.7/24";
+                    macAddress = "02:00:00:03:03:07";
+                    ramMb = 512;
+                    cores = 1;
+                  }
```

Comment:
can be fixed in a later patch, but should these definitions be in modules/vms/ or something similar, so that they could be more easily shared ?

Reply:
SP-2840 is created to track this